src/builder/pod/mod.rs

```diff
@@ -156,6 +157,19 @@ impl PodBuilder {
         self
     }
 
+    /// Extend the pod's image_pull_secrets field with the pull secrets from a given [ResolvedProductImage]
+    pub fn image_pull_secrets_from_product_image(
```

Comment:
Honestly I don't think I have a better way to do this.. but it feels awkward that you need to call both `image_from` and `image_pull_secrets_from` to get the correct behaviour, with no warning that you've done it wrong.

Reply:
Agreed, but i also don't know a better way of doing this